### DISCUSSION TITLE
[RFC] Reinforcement learning from human feedback

### DISCUSSION THREAD

**Discussion by pppppM** [InternLM/xtuner#257] [Ideas]
Discuss the Implementation of RLHF

**Comment by xiaohangguo** [comment]
# Paper Reading List
- [Aligning Language Models with Offline Reinforcement Learning from Human
Feedback](https://arxiv.org/pdf/2308.12050.pdf)
- [Training language models to follow instructions with human feedback](https://arxiv.org/pdf/2203.02155.pdf)
- [Improving alignment of dialogue agents via targeted human judgements](https://arxiv.org/pdf/2209.14375.pdf) 
- [Fine-Tuning Language Models from Human Preferences](https://arxiv.org/abs/1909.08593)
- [DeepSpeed-Chat: Easy, Fast and Affordable RLHF
Training of ChatGPT-like Models at All Scales](https://arxiv.org/pdf/2308.01320.pdf)
---------------
选了几篇比较经典RLHF（Reinforcement learning from human feedback）论文

**Comment by maxchiron** [comment]
大佬慢点，我先入个门😭
[https://pytorch.org/tutorials/intermediate/reinforcement_q_learning.html](url)

**Comment by helonggood** [comment]
我认领这篇：Aligning Language Models with Offline Reinforcement Learning from Human Feedback，研究研究，到时候分享一个ppt

**Reply by helonggood** [reply]
【腾讯文档】RLHF_OFFLINE_PPT
https://docs.qq.com/slide/DZmJLbkp4aUZTWmJT
分享PPT，大家有兴趣可以阅览讨论。
文章研究摘要：
1、文章提出一个来自人类反馈 (RLHF) 框架的离线强化学习，以使用生成的样本对齐 LM，而无需与 RL 环境交互。
2、主要研究了如下三种离线RLHF方法来对齐人类偏好：maximum likelihood estimation (MLE) with filtering, reward-weighted regression (RWR), and Decision Transformer (DT) 。
3、通过采用类似监督学习的损失函数，结果表明这三种方法相较于PPO，优势体现在：机器学习系统 (MLSys)更简单 、更少的（大约 12.3%）计算资源、更稳定的模型训练。
4、同时，DT 对齐优于其他离线 RLHF 方法，也优于 PPO，结论就是DT方法最好。

**Comment by Sun-9923** [comment]
我认领这篇：DeepSpeed-Chat: Easy, Fast and Affordable RLHF Training of ChatGPT-like Models at All Scales

**Comment by xiaohangguo** [comment]
我认领openAI这篇吧：《Training language models to follow instructions with human feedback》，到时候一起ppt圆桌分享哇。

**Comment by maxchiron** [comment]
我认领这篇：[Fine-Tuning Language Models from Human Preferences](https://arxiv.org/abs/1909.08593)

**Comment by amulil** [comment]
我认领这篇：[Improving alignment of dialogue agents via targeted human judgements](https://arxiv.org/pdf/2209.14375.pdf)

**Comment by KooSung** [comment]
考虑支持视觉的[RLHF-V](https://arxiv.org/abs/2312.00849)吗？
code: https://github.com/RLHF-V/RLHF-V, https://github.com/thunlp/Muffin
